pyccel/ast/core.py

```diff
@@ -4669,29 +4669,27 @@ def get_assigned_symbols(expr):
 
         try:
             var = expr.lhs
-
+            symbols = []
             if isinstance(var, DottedVariable):
                 var = expr.lhs.lhs
                 while isinstance(var, DottedVariable):
                     var = var.lhs
+                symbols.append(var)
             elif isinstance(var, IndexedElement):
                 var = var.base
-            free_symbols = var.free_symbols
-            symbols = list(free_symbols)
+                symbols.append(var)
+            return symbols
         except:
 
             # TODO must raise an Exception here
             #      this occurs only when parsing lapack.pyh
```

Comment:
@saidctb Here the TODO suggests to raise an `Exception`, but you added a `pass` statement instead. Why?

Reply:
I don't know what @ratnania  has in mind here so I didn't do anything

Reply:
Have you tried raising an Exception and see if we run into any problems?